vagrant.go

```diff
@@ -300,21 +316,45 @@ func (v *Vagrant) vagrantCommand(args ...string) *exec.Cmd {
 func (v *Vagrant) runVagrantCommand(args ...string) (string, error) {
 	cmd := v.vagrantCommand(args...)
 	cmd.Dir = v.VagrantfilePath
+
+	v.debugf("executing: %v", cmd.Args)
+
 	out, err := cmd.CombinedOutput()
 	if err != nil {
 		if len(out) != 0 {
 			err = fmt.Errorf("%s: %s", err, out)
 		}
-		return "", err
+		return "", v.error(err)
 	}
 
+	v.debugf("%s", out)
+
 	return string(out), nil
 }
 
+func (v *Vagrant) debugf(format string, args ...interface{}) {
+	if v.Log != nil {
+		v.Log.New(v.VagrantfilePath).Debug(format, args...)
+	}
+}
+
+func (v *Vagrant) errorf(format string, args ...interface{}) error {
+	err := fmt.Errorf(format, args...)
+	v.debugf("%s", err)
+	return err
+}
+
+func (v *Vagrant) error(err error) error {
+	if err != nil {
+		v.debugf("%s", err)
+	}
+	return err
+}
+
 // startCommand starts the command and sends back both the stdout and stderr to
 // the returned channel. Any error happened during the streaming is passed to
 // the Error field.
-func startCommand(cmd *exec.Cmd) (<-chan *CommandOutput, error) {
+func (v *Vagrant) startCommand(cmd *exec.Cmd) (<-chan *CommandOutput, error) {
```

Comment:
Same for this, no need to make it a method of `*Vagrant`. It would make it easier to implement tests